include/nbl/builtin/hlsl/sort/common.hlsl

```diff
@@ -0,0 +1,31 @@
+// Copyright (C) 2018-2024 - DevSH Graphics Programming Sp. z O.O.
+// This file is part of the "Nabla Engine".
+// For conditions of distribution and use, see copyright notice in nabla.h
+
+#ifndef _NBL_BUILTIN_HLSL_SORT_COMMON_INCLUDED_
+#define _NBL_BUILTIN_HLSL_SORT_COMMON_INCLUDED_
+
+namespace nbl
+{
+namespace hlsl
+{
+namespace sort
+{
+
+struct CountingPushData
+{
+    uint64_t inputKeyAddress;
+    uint64_t inputValueAddress;
+    uint64_t scratchAddress;
+    uint64_t outputKeyAddress;
+    uint64_t outputValueAddress;
+    uint32_t dataElementCount;
+    uint32_t minimum;
+    uint32_t elementsPerWT;
+};
```

Comment:
imho should be templated on types `K` and `V` and using `bda::__ptr` instead of plain `uint64_t` so we know what we're sorting. with a static assert or something that `V` should be `is_integral`

Also we don't know we're using push cosntants for this at this stage so just call it `CountingParameters` and move into the `counting.hlsl` header so everything is in one place

Also `dataElementCount` and `elementsPerWT` could be packed into a single `uint64_t` bitfield (with `dataElementCount` getting a bit more bits than 32, maybe 36)

And `minimum` needs to be of type `K`

Reply:
Actually none of the addresses should be here, the accessors you pass to the counting sort should already know about them and handle them as state.

Only thing you need here is the element count, minimum key value (so template only on `V`) and elements per invocation

Reply:
also if you have a minumum you should also have a maximum or bucket count

Reply:
Done